engines/engine.cpp

```diff
@@ -493,13 +493,23 @@ void Engine::handleAutoSave() {
 
 void Engine::saveAutosaveIfEnabled() {
 	if (_autosaveInterval != 0) {
-		bool canSave = canSaveAutosaveCurrently();
+		bool saveFlag = canSaveAutosaveCurrently();
+		Common::String saveName = _("Autosave");
+
+		if (saveFlag) {
+			// First check for an existing savegame in the slot, and if present, if it's an autosave
+			SaveStateDescriptor desc = getMetaEngine().querySaveMetaInfos(
+				_targetName.c_str(), getAutosaveSlot());
+			saveFlag = desc.getSaveSlot() == -1 || desc.getDescription() == saveName;
```

Comment:
Is testing on the save name really a good idea? What if the user changes the language of the GUI?

Reply:
There is that possibility, yes. But testing the name is the best solution I could come up. I think still allowing for the 'Autosave' to be translateable is better for foreign language users then forcing the text to always be English.

Reply:
Have you considered adding a `isAutoSave` flag to `SaveStateDescriptor`? Engines that save the information that a save is an autosave could valorize it. For the other engines it could default to the description being `Autosave`. That way, newer engines can be more robust.

Reply:
The trouble with that is all the different historical engines provide their own querySaveMetaInfos method implementations. So even if I added an isAutosave property to SaveStateDescriptor, I'd have to add the comparison to "Autosave" to every one of them to set it as necessary. Which is too much code duplication for my liking.

Reply:
What about:
```c++
enum SaveType {
    kSaveTypeUndetermined,
    kSaveTypeRegular,
    kSaveTypeAutosave
};

SaveType _saveType { kSaveTypeUndetermined };

void setAutosave(bool autosave) {
    _saveType = autosave ? kSaveTypeAutosave : kSaveTypeRegular;
}

bool isAutosave() const {
    if (_saveType != kSaveTypeUndetermined) {
        return _saveType == kSaveTypeAutosave;
    } else {
        return _description == _("Autosave");
    }
}
```

Reply:
Now that I like. And an autosave property could also be fairly easily added to the extended savegame format to as well, which could set the property.